File: 0268_missing_number/missing_number.cpp
#include <iostream>
#include <vector>
#include <numeric>

using namespace std;

/**
 * Time: O(n)
 * Space: O(1)
 */
int missing_number(vector<int> &nums)
{
    int missing = nums.size() * (nums.size() + 1) / 2;
    return missing - accumulate(nums.begin(), nums.end(), 0);
}

File: 0268_missing_number/missing_number.go
package main

/**
 * Time: O(n)
 * Space: O(1)
 */
func missingNumber(nums []int) int {
	missing := len(nums)
	for i, e := range nums {
		missing ^= i ^ e
	}
	return missing
}

func main() {

}
